opacus/optimizers/perlayeroptimizer.py

```diff
@@ -51,13 +52,16 @@ def __init__(
             loss_reduction=loss_reduction,
             generator=generator,
             secure_mode=secure_mode,
+            ew_compatibility_mode=ew_compatibility_mode,
         )
 
     def clip_and_accumulate(self):
         for (p, max_grad_norm) in zip(self.params, self.max_grad_norms):
             _check_processed_flag(p.grad_sample)
 
-            per_sample_norms = p.grad_sample.view(len(p.grad_sample), -1).norm(2, dim=1)
+            per_sample_norms = p.grad_sample.reshape(len(p.grad_sample), -1).norm(
```

Comment:
Same here: no need for reshape before computing norm.

Reply:
Indeed, will update, thanks!